teamcity/resource_build_config.go

```diff
@@ -630,6 +676,59 @@ func flattenBuildStepCmdLine(s *api.StepCommandLine) map[string]interface{} {
 	return m
 }
 
+func flattenBuildFeatures(bfs []api.BuildFeature) ([]map[string]interface{}, error) {
+	var bfsToSave []map[string]interface{}
+	for _, bf := range bfs {
+		bfToSave := make(map[string]interface{})
+		gbf := bf.(*api.GenericBuildFeature)
+		bfToSave["type"] = gbf.Type()
+
+		props := gbf.Properties()
+		if props != nil && props.Count > 0 {
+			bfToSave["properties"] = make(map[string]string)
```

Comment:
Perhaps this line should be moved to outside the if-statement. I think an empty map will probably be less error-prone than if the flattenedBuildFeatures is missing the `"properties"` key.